samtranslator/translator/translator.py

```diff
@@ -269,6 +272,80 @@ def _get_resources_to_iterate(
+    connectors = {}
+
+    # Loop through the resources in the template and see if any connectors have been attached
+    for source_logical_id, resource in resources.items():
+        if "Connectors" in resource:
+            if not isinstance(resource["Connectors"], dict):
+                raise_error("'Connectors' must be a dict")
+
+            # Go through each of the connectors that have been attached and create a Serverless Connector resource
+            for connector_logical_id, connector_dict in resource["Connectors"].items():
+                if not isinstance(connector_dict, dict):
+                    raise_error(f"Invalid resource type. '{connector_logical_id}' be a dict.")
+
+                # Make sure there are no other embedded connectors with the same logical id
+                if connector_logical_id in connectors:
+                    raise_error(
+                        f"Another connector with the logical id {connector_logical_id} exists in the "
+                        f"template. Please rename either of them."
+                    )
+
+                # mutates the connector_dict to make it a connector resource
+                get_generated_connector(source_logical_id, connector_logical_id, connector_dict)
+                connectors[connector_logical_id] = connector_dict
+
+            # delete connectors key, so it doesn't show up in the transformed template
+            del resource["Connectors"]
+
+    # go through the list of generated connectors and add it to resources dict
+    for connector_logical_id, connector_dict in connectors.items():
+        if connector_logical_id in resources:
+            raise_error(
+                f"The logical ID of the generated connector {connector_logical_id} has a conflict with "
+                f"{connector_logical_id} of type '{resources[connector_logical_id]['Type']}', "
+                f"please rename either of them "
+            )
+        resources[connector_logical_id] = connector_dict
+
+
+def get_generated_connector(source_logical_id: str, connector_logical_id: str, connector: Dict[str, Any]) -> None:
+    """
+    Adds the type, source id and any ResourceReference properties to the connector
+
+    :param str source_logical_id: Logical id of the resource the connector is attached to
+    :param str connector_logical_id: Logical id of the connector
+    :param dict connector: the properties of the connector including the Destination, Permissions and SourceReference [Optional]
+    """
+    connector_class = SamConnector(logical_id="SamConnector")
+    connector["Type"] = connector_class.resource_type
+    properties = connector["Properties"]
+
+    # No need to raise an error for this instance as the error will be caught by the parser
+    if isinstance(properties, dict):
+        properties["Source"] = {"Id": source_logical_id}
+
+        if "SourceReference" in properties:
+            if not isinstance(properties["SourceReference"], dict):
```

Comment:
These are the properties complementing what's deduced from the source resource? If yes, that's not very clear from the name, which seems to suggest a fully-defined reference to the source resource. I don't have any good suggestions right now that's not mega verbose like `AdditionalSourceProperties`. 🤔

Reply:
The UX team preferred `SourceReference` (at least they preferred it over using just `Source`) but we can change it to something clearer. Maybe `SourceResourceReference`? Customers should be familiar with the `ResourceReference` if they read the docs and adding Source as a prefix could minimize confusion.